src/types.rs

```diff
@@ -167,3 +171,35 @@ pub struct GetNetworkInfoResult {
     pub local_addresses: Vec<GetNetworkInfoResultAddress>,
     pub warnings: String,
 }
+
+#[derive(Serialize, Clone, Eq, PartialEq, Debug, Hash)]
+pub struct DescriptorChecksum(pub String);
+
+impl FromStr for DescriptorChecksum {
+    type Err = BwtError;
+    fn from_str(s: &str) -> Result<DescriptorChecksum, Self::Err> {
+        Ok(DescriptorChecksum(s.to_string()))
+    }
+}
+
+#[derive(Serialize, Clone, Eq, PartialEq, Debug, Hash)]
+pub struct Descriptor(pub String);
+
+impl Descriptor {
+    pub fn new(descriptor: &str, rpc: &RpcClient) -> Result<Self, BwtError> {
+        // TODO: what to do about non-ranged descriptors?
```

Comment:
I would leave non-ranged descriptor aside for now, the logic for importing them is quite different compared to ranged ones. Importing single addresses is currently not supported either, so its okay if this PR doesn't support it initially.